integration-tests/tests/src/utils/open-realm.ts

```diff
@@ -38,43 +38,60 @@ export async function openRealm(
   partialConfig: LocalConfiguration | SyncedConfiguration = {},
   user: User,
 ): Promise<{ config: Configuration; realm: Realm }> {
-  const nonce = new BSON.ObjectId().toHexString();
-  const path = `temp-${nonce}.realm`;
-
   if (!partialConfig.sync) {
-    const config = { ...partialConfig, path } as LocalConfiguration;
+    const config = createLocalConfig(partialConfig as LocalConfiguration);
     const realm = await Realm.open(config);
     return { config, realm };
-  } else {
-    const config = {
-      ...partialConfig,
-      path,
-      sync: {
-        user: user,
-        ...(partialConfig.sync.flexible ? { flexible: true } : { partitionValue: nonce }),
-        _sessionStopPolicy: "immediately",
-        ...partialConfig.sync,
-      },
-    } as Configuration;
-    const realm = await Realm.open(config);
+  }
 
-    // Upload the schema, ensuring a valid connection. uploadAllLocalChanges
-    // will not resolve with flexible sync enabled until we have created an
-    // initial subscription set, so skip it if we have a flexible config.
-    // There is an issue on the cloud side which will fix this.
-    if (!config.sync?.flexible) {
-      if (!realm.syncSession) {
-        throw new Error("No syncSession found on realm");
-      }
+  const config = createSyncConfig(partialConfig, user);
+  const realm = await Realm.open(config);
 
-      await realm.syncSession.uploadAllLocalChanges();
+  // Upload the schema, ensuring a valid connection. uploadAllLocalChanges
+  // will not resolve with flexible sync enabled until we have created an
+  // initial subscription set, so skip it if we have a flexible config.
+  // There is an issue on the cloud side which will fix this.
+  if (!config.sync?.flexible) {
+    if (!realm.syncSession) {
+      throw new Error("No syncSession found on realm");
     }
 
-    // TODO: This should probably be done in Realm.open()
-    if (config.sync?.flexible) {
-      await realm.subscriptions.waitForSynchronization();
-    }
+    await realm.syncSession.uploadAllLocalChanges();
+  }
 
-    return { config, realm };
+  // TODO: This should probably be done in Realm.open()
+  if (config.sync?.flexible) {
+    await realm.subscriptions.waitForSynchronization();
   }
+
+  return { config, realm };
+}
+
+export function createSyncConfig(partialConfig: SyncedConfiguration = {}, user: User): Configuration {
+  const { path, nonce } = getRandomPathAndNonce();
+
+  return {
+    ...partialConfig,
+    path,
+    sync: {
+      user: user,
+      ...(partialConfig.sync?.flexible ? { flexible: true } : { partitionValue: nonce }),
+      _sessionStopPolicy: "immediately",
+      ...partialConfig.sync,
+    },
+  } as unknown as Configuration;
+}
+
+export function createLocalConfig(partialConfig: LocalConfiguration = {}): Configuration {
+  const path = getRandomPathAndNonce().path;
+
+  return { ...partialConfig, path };
+}
+
+function getRandomPathAndNonce(): { path: string; nonce: string } {
```

Comment:
In master we have a file `/src/utils/generators.ts` that deals with generating random values. Think it might make sense to move this method to it. If so it could make sense to add a TODO at least for this PR (not sure if you have the necessary changes from master since it require the bindgen rebase to be done).

Reply:
Added a note, thanks!